File: backend/app/models/student.py
import uuid
from datetime import datetime
from sqlalchemy import String, Integer, DateTime
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.core.database import Base

class Student(Base):
    __tablename__ = "students"

    id: Mapped[str] = mapped_column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    student_code: Mapped[str] = mapped_column(String, unique=True, index=True, nullable=False)
    first_name: Mapped[str] = mapped_column(String, nullable=False)
    last_name: Mapped[str] = mapped_column(String, nullable=False)
    age: Mapped[int] = mapped_column(Integer, nullable=False)
    gender: Mapped[str] = mapped_column(String, nullable=False) # 'M' or 'F'
    school: Mapped[str] = mapped_column(String, nullable=False) # 'GP' or 'MS'

    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
    updated_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    # Relationships
    academic_records = relationship("AcademicRecord", back_populates="student", cascade="all, delete-orphan")
    documents = relationship("Document", back_populates="student", cascade="all, delete-orphan")
    predictions = relationship("Prediction", back_populates="student", cascade="all, delete-orphan")

File: frontend/src/types/student.ts
export interface Student {
  id: string;
  student_code: string;
  first_name: string;
  last_name: string;
  age: number;
  gender: 'M' | 'F';
  school: 'GP' | 'MS';
  created_at: string;
  updated_at: string;
}

export interface AcademicRecord {
  id: string;
  student_id: string;
  study_time: number;
  failures: number;
  absences: number;
  family_support: 'yes' | 'no';
  school_support: 'yes' | 'no';
  internet_access: 'yes' | 'no';
  health: number;
  g1: number;
  g2: number;
  created_at: string;
}
